src/main/java/com/joe/example/hibernate_map/test/entity/Asset.java

```diff
@@ -17,7 +17,7 @@
     private BigDecimal price;
 
     @OneToOne(cascade = CascadeType.ALL)
-    @JoinColumn(name = "asset_id", referencedColumnName = "asset_id", nullable = false, insertable = false, updatable = false, foreignKey = @ForeignKey(ConstraintMode.NO_CONSTRAINT))
+    @JoinColumn(name = "asset_detail_id", referencedColumnName = "asset_id",  insertable = false,foreignKey = @ForeignKey(ConstraintMode.NO_CONSTRAINT))
```

Comment:
Thanks @dreab8, this change does work. But looks like it creates this "_hidden_" **asset_detail_id** column in the asset table. So this is new one, I've also found other ways to make it works.. e.g: going with standard @Id path instead of @EmbeddedId or using bi-directional instead of uni-directional.

Is seems that for uni-directional OneToOne relationship, hibernate expects both entities's id class to be the same type? Would you know anything about that? (When I make AssetId to be the embeddable id of both Asset and AssetDetail, the test passes.)

Btw, thanks for sharing that link. I found it useful.